src/components/WizardFormProvider/WizardFormProvider.tsx

```diff
@@ -1,27 +1,31 @@
 import { ReactNode, useState } from 'react';
 import FormScreens from 'constants/formScreens';
-import initialContextValue from 'constants/initialContextValue';
-import WizardFormContext from 'contexts/WizardFormContext';
+import WizardFormContext, {
+  initialFormValues,
+} from 'contexts/WizardFormContext';
 import PersonalDetailsValues from 'types/personalDetailsValues';
 
 interface Props {
   children: ReactNode;
 }
 
-interface FormValues {
-  formValues: { [FormScreens.PERSONAL_DETAILS]: PersonalDetailsValues };
-  setFormValues: (values: PersonalDetailsValues) => {
-    [FormScreens.PERSONAL_DETAILS]: PersonalDetailsValues;
-  };
-}
-
 export default function WizardFormProvider({ children }: Props) {
-  const [formValues, setFormValues] = useState<FormValues>(initialContextValue);
+  const [formValues, setFormValues] =
+    useState<typeof initialFormValues>(initialFormValues);
 
+  const onSelectFormValues = (
+    screenName: FormScreens,
```

Comment:
You need to use generic here that extends enum `FormScreens`. `screen` (pls, rename it to this name) should have type of this generic and values should use generics for indexed access type